src/main/java/org/sunbird/workflow/service/impl/UserProfileWfServiceImpl.java

```diff
@@ -64,11 +69,122 @@ public void updateUserProfile(WfRequest wfRequest) {
 	}
 
 	private void updateProfile(WfRequest wfRequest) {
+		final ObjectMapper mapper = new ObjectMapper();
 		StringBuilder builder = new StringBuilder();
-		String endPoint = configuration.getHubProfileUpdateEndPoint().replace(Constants.USER_ID_VALUE,
+		String endPoint = configuration.getUserProfileReadEndPoint().replace(Constants.USER_ID_VALUE,
 				wfRequest.getApplicationId());
-		builder.append(configuration.getHubServiceHost()).append(endPoint);
-		requestServiceImpl.fetchResult(builder, wfRequest.getUpdateFieldValues(), Map.class);
+		builder.append(configuration.getLmsServiceHost()).append(endPoint);
+		Object readResponse = requestServiceImpl.fetchResultUsingGet(builder);
+		Map<String, Object> readData = mapper.convertValue(readResponse,Map.class);
```

Comment:
move this USER READ API handling as separate method -- so that we can read it easily...